### PR TITLE
Interactive search video on youtube

### PR DESCRIPTION
**Pull request by Eiphoria**

### CODE REVIEW

File: bot/errors/filter.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class FilterStrLenError(Exception):
+    """Exception raised for errors in the input value.
```
Comment (Drozd0f) [inline bot/errors/filter.py]: delete docstring

File: bot/errors/filter.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+    Attributes:
+        message -- explanation of the error
+    """
+
+    def init(self, message='Input lenght of srting is too long'):
```
Comment (Drozd0f) [inline bot/errors/filter.py]: you not use message in later delete this

File: bot/events/users/play.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+log = logging.getLogger(__name__)
+
+
+async def get_link(message: Message, links: t.List[str], titles: t.List[str]) -> t.Optional[str]:
+    num = int(message.content)
+    if num in range(1, Config.result_len + 1):
+        await message.channel.send(f'You choose {num} - {titles[num-1]}:{links[num-1]}')
+        return links[num-1]
+    await message.channel.send(
+        f'Abort! :anger: You choose {num} - which is not valid,'
```
Comment (Drozd0f) [inline bot/events/users/play.py]: May be better use dataclass DefaultEmoji in src/emoji.py and this class will be contains default emoji
```pyhon
class DefaultEmoji:
    anger = :anger:
```

And then
```pyhon
f'Abort! {DefaultEmoji.anger} You choose {num} - which is not valid,'
```

It's better for understanding what you mean by string `:anger:`

File: bot/src/yt_link_changer.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    def link_rebaser(self, link: str, regex: str) -> str:
+        return re.sub(regex, self.base_str, link)
+
+    def yt_search(self, text: str, res_len: int = 3) -> str:
+        videosSearch = VideosSearch(text, limit=res_len)
+        search_result = videosSearch.result()
+        video_titles = []
+        video_links = []
+        for idx, res in enumerate(search_result.get('result')):
+            video_titles.append(f"{idx + 1}. {res.get('title')} \n")
```
Comment (Drozd0f) [inline bot/src/yt_link_changer.py]: delete `\n` and space befor them

File: bot/events/users/play.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        f'type number only in range(1-{Config.result_len}) :anger:'
+    )
+
+
+async def link_extractor(bot: Bot, message: Message, request_proc: YTLinks, text: str) -> t.Optional[str]:
+    titles, links = request_proc.yt_search(text, Config.result_len)
+    choice_dialog = f'Type num of the song 1-{Config.result_len} (example: 2) \n' + ' '.join(titles)
```
Comment (Drozd0f) [inline bot/events/users/play.py]:
```python
' '.join(titles)
```
Transform to
```python
'\n'.join(titles)
```

File: bot/events/users/play.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    await message.channel.send(choice_dialog)
+
+    def is_my_message(msg):
```
Comment (Drozd0f) [inline bot/events/users/play.py]: ```is_my_message``` -> ```is_author```

File: bot/events/users/play.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        return msg.author == message.author and msg.channel == message.channel
+    user_message = await bot.wait_for('message', check=is_my_message)
+
+    try:
+        return await get_link(user_message, links, titles)
+    except ValueError:
+        await message.channel.send(
+            f'{user_message.content} mean that hoisting crane, but you are adopted :scream_cat: ))'
```
Comment (Drozd0f) [inline bot/events/users/play.py]: *`:scream_cat:` - default emoji too

File: bot/events/users/play.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+
+def play(bot: Bot):
+    async def on_message(message: Message):
+        if message.content.startswith('?play'):
+            command, *text = message.content.split()
+            text = ' '.join(text)
+            request_proc = YTLinks()
+            try:
+                result = request_proc.filter(text)
```
Comment (Drozd0f) [inline bot/events/users/play.py]: `result` -> `url`

File: bot/events/users/play.py [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+            except filter.FilterStrLenError:
+                await message.channel.send(f'Query lenght is too long(more than {Config.query_len} symbols)')
+                return
+            if not result:
+                result = await link_extractor(bot, message, request_proc, text)
```
Comment (Drozd0f) [inline bot/events/users/play.py]: if you use request_proc in this func maybe implement this in YTLinks?

File: bot/src/yt_link_changer.py [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+            video_links.append(f"{res.get('link')}")
+        return video_titles, video_links
+
+    def filter(self, text: str) -> t.Optional[str]:
+        if re.match(self.watch_link, text):
+            return self.link_rebaser(text, self.watch_link)
+        elif re.match(self.shorts_link, text):
+            return self.link_rebaser(text, self.shorts_link)
+        elif re.match(self.base_link, text):
+            return text
```
Comment (Drozd0f) [inline bot/src/yt_link_changer.py]: implement there `link_extractor`

**Review by Drozd0f — CHANGES_REQUESTED** [review]

File: bot/events/users/play.py (hunk 9 above)
Comment (Eiphoria) [reply bot/events/users/play.py]: Unga not bunga
image

File: bot/src/yt_link_changer.py (hunk 10 above)
Comment (Eiphoria) [reply bot/src/yt_link_changer.py]: image
Unga don't definitely understand what is meaning there

File: bot/errors/filter.py [hunk 11]
```diff
@@ -0,0 +1,4 @@
+class FilterStrLenError(Exception):
+
+    def init(self):
```
Comment (Drozd0f) [inline bot/errors/filter.py]: delete this :)

File: bot/src/yt_link_changer.py [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+        elif re.match(self.shorts_link, text):
+            url = self.link_rebaser(text, self.shorts_link)
+        elif re.match(self.base_link, text):
+            url = text
+        else:
+            url = await self.link_extractor(bot, message, text)
+        return f'{command} {url}'
+
+    @staticmethod
+    async def get_link(message: Message, links: t.List[str], titles: t.List[str]) -> t.Optional[str]:
```
Comment (Drozd0f) [inline bot/src/yt_link_changer.py]: move staticmethod to the end of class

File: bot/__init__.py [hunk 13]
```diff
@@ -30,5 +31,8 @@ def start_bot():
     log.info("setuping cogs...")
     register_all_cogs(BOT)
 
+    log.info("setuping events...")
```
Comment (Drozd0f) [inline bot/__init__.py]: Let's decide which quotes to use. Because in one pull request single and double quotes.

**Review by Drozd0f — CHANGES_REQUESTED** [review]

**Review by Drozd0f — APPROVED** [review]

**Pull request merged**